xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2156,6 +2156,7 @@ void CVideoPlayer::CheckContinuity(CCurrentStream& current, DemuxPacket* pPacket
        fabs(this_dts - that_dts) < DVD_MSEC_TO_TIME(1000))
     {
       m_offset_pts += correction;
+      m_VideoPlayerVideo->SetOffsetPts(m_offset_pts);
```

Comment:
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDDemuxSPU.cpp#L208

It does not need to correct anything because CDVDDemuxSPU::AddData gets the corrected pts already